### ISSUE TITLE
feat: stage points-left-on-the-table breakdown (hit quality vs penalties)

### ISSUE THREAD

**Issue by mandakan** [mandakan/ssi-scoreboard#46]
## Classification

- **Category**: Coaching / post-match — most actionable after all stages are fired; adding this data mid-match adds noise without decision value.
- **UI zone**: Collapsible "Analysis" panel below the comparison table, hidden by default. Gate behind a "Show coaching data" toggle or only render when the majority of stages have been fired. Must not appear in the default courtside comparison view.

## Coaching value

Knowing the total points gap is useful; knowing *why* the gap exists is actionable. This feature separates lost points into two root causes:

1. **Hit quality** — Cs, Ds, and misses where an A was possible
2. **Penalties** — misses scored as misses, no-shoots, procedurals

A shooter whose losses are mostly penalties needs mental-game work; one whose losses are mostly hit quality needs marksmanship/sight-picture work.

## What to build

For each stage and competitor, given the raw scorecard counts `(a, c, d, miss, ns, proc)`:

```
a_max        = (a + c + d + miss + ns_hits) × A_value   // best possible points on engaged targets
hit_loss     = a_max − points_from_A_C_D_only
penalty_loss = abs(points_from_penalties)                // misses, NS, procedurals (already negative)
total_loss   = hit_loss + penalty_loss
```

A collapsed summary badge (e.g. `−18 pts to penalties`) in the totals row links to the expanded panel. The full breakdown lives in a collapsible analysis section, not inline in the comparison table.

Match-level stacked horizontal bar per shooter showing: hit-quality loss / penalty loss / remaining points.

## Implementation hints

- The scorecard data (`a`, `c`, `d`, `miss`, `ns`, `proc`, `points`) is already fetched by the compare API — extend `computeGroupRankings()` to derive `hitLossPoints` and `penaltyLossPoints` per stage entry.
- Add both fields to the relevant interface in `lib/types.ts`.
- Minor/major power factor affects `A_value`, `C_value`, etc. — handle both cases.
- Share penalty totals with the penalty rate feature to avoid duplicate computation.
- Unit-test the loss computation in `tests/unit/` with edge cases (all As, all misses, procedural-only penalties).

## Acceptance criteria

- [ ] Collapsible Analysis panel is hidden by default on all viewports
- [ ] Summary badge visible in totals row when losses are non-zero
- [ ] Per-stage breakdown and match-level stacked bar visible when panel is expanded
- [ ] Power factor (major / minor) handled correctly
- [ ] `pnpm typecheck && pnpm test` pass with zero errors/warnings
- [ ] Pure computation logic is unit-tested
